### ISSUE TITLE
Switch Json dependency from gwt-elemental to jsonp

### ISSUE THREAD

**Issue by stbischof** [vaadin/flow#10230]
Would it be an option to switch from  `gwt-elemental/elemental.json` to [JsonP](https://jakarta.ee/specifications/jsonp/2.0/) to keep the dependencies a cleaner and importing just an API and not an implementation?

**Comment by Legioth** [comment]
The reason we're using the slightly clunky gwt-elemental dependency is that JSON objects are `Serializable` and can thus be stored inside the HTTP session without workarounds. We evaluated various alternatives such as Jackson, Gson and JSON-P but neither of them have this critical feature.

**Comment by stbischof** [comment]
@Legioth thank you for the fast answer! Good to know that you evaluated this.

I'm am not sure that i understand it correct because is did not see an `implements Serializable` in elemental.json.

But in jackson an
```
public class JSONPObject
    implements JacksonSerializable
{
```

It would help me a lot to understand is you may show the code /test where this Serialization happened, that you mean.

**Comment by Legioth** [comment]
I see. That fork is no longer maintained.

**Comment by Artur-** [comment]
Cool, seems like it was added in 2.10: https://cowtowncoder.medium.com/jackson-2-10-feature-jsonnode-improvements-18894c3ac3b5

**Comment by jhult** [comment]
Is this a duplicate of #8218?

**Comment by Artur-** [comment]
Not exactly. #8218 is about the public facing API while this is about the internals. It is possible to change the public API and keep gwt-elemental in use internally
